### PR TITLE
Android feature/items styling

### PR DESCRIPTION
**Pull request by AWhiteSheep**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ install:
     - npm install
     - grunt compile
 
+after_failure:
```
Comment (PeterStaev) [inline .travis.yml]: I dont think we need this. The error is already clearly visible in the grunt/npm task fails. This just clutters the output.

File: README.md [hunk 2]
```diff
@@ -65,6 +65,12 @@ Gets or sets whether the drop down is enabled. If you want to apply a specific s
 * **accessoryViewVisible** - *boolean* (Default: true)  
 Gets/sets whether there will be an accessory view (toolbar with Done button) under iOS. Valid only when running on iOS.
 
+* **itemsTextAlignment** - *String*
+Gets or sets the alignment for items in the DropDown. eg: `itemsTextAlignment="center"`.
```
Comment (PeterStaev) [inline README.md]: no need for `eg:...`. this can be viewed from the demo usage below.

File: drop-down-common.ts [hunk 3]
```diff
@@ -154,4 +156,16 @@ export const hintProperty = new Property<DropDownBase, string>({
     name: "hint",
     defaultValue: ""
 });
-hintProperty.register(DropDownBase);
\ No newline at end of file
+hintProperty.register(DropDownBase);
+
+export const itemsTextAlignmentProperty = new Property<DropDownBase, string>({
```
Comment (PeterStaev) [inline drop-down-common.ts]: this property should be changed. Its type should be `TextAlignment` not string. Also it should have a `valueConverter` and default value should be `initial`. Take a look at how it is done in the NS core modules:
https://github.com/NativeScript/NativeScript/blob/46705ee33278e8f9ec04b00ea592dd65ba855e74/tns-core-modules/ui/text-base/text-base-common.ts#L198-L200

File: README.md [hunk 4]
```diff
@@ -65,6 +65,12 @@ Gets or sets whether the drop down is enabled. If you want to apply a specific s
...
+
+* **itemsPadding** - *String* android only! for now
```
Comment (PeterStaev) [inline README.md]: I prefer to not have properties supporting only one of the OS-es. So let's keep this PR for the `itemsTextAlignment` and when you get `itemsPadding` working for iOS you can submit another one with just that?

File: drop-down.android.ts [hunk 5]
```diff
@@ -409,13 +425,36 @@ function initializeDropDownAdapter() {
                     label.style.placeholderColor = owner.style.placeholderColor;
                 }
                 label.style.textDecoration = owner.style.textDecoration;
-                label.style.textAlignment = owner.style.textAlignment;
+                const _textAlignment = {
```
Comment (PeterStaev) [inline drop-down.android.ts]: Once the type of the property in the common file is changed to `TextAlignment` there will not be needed. And you can directly assign it to the `label.style.textAlignment`

File: drop-down.android.ts [hunk 6]
```diff
@@ -409,13 +425,36 @@ function initializeDropDownAdapter() {
...
+                    _leftAlignment: "left" as TextAlignment,
+                    _centerAlignment: "center" as TextAlignment,
+                    _rightAlignment: "right" as TextAlignment
+                };
+                if (owner.itemsTextAlignment !== "" && realizedViewType === 1) {
```
Comment (PeterStaev) [inline drop-down.android.ts]: should be `!== "initial"`

File: drop-down.d.ts [hunk 7]
```diff
@@ -35,6 +35,8 @@ export class DropDown extends View {
 
     public items: any[] | ItemsSource;
     public selectedIndex: number;
+    public itemsTextAlignment: string;
```
Comment (PeterStaev) [inline drop-down.d.ts]: type should be `TextAlignment`

File: drop-down.ios.ts [hunk 8]
```diff
@@ -191,6 +192,13 @@ export class DropDown extends DropDownBase {
         this.ios.hint = value;
     }
 
+    public [itemsTextAlignmentProperty.getDefault](): string {
+        return "";
```
Comment (PeterStaev) [inline drop-down.ios.ts]: should be `"initial"`

File: drop-down.ios.ts [hunk 9]
```diff
@@ -362,8 +370,11 @@ class DropDownListPickerDelegateImpl extends NSObject implements UIPickerViewDel
         };
 
         label.font = style.fontInternal.getUIFont(label.font);
-
-        switch (style.textAlignment) {
+        let p_itemsTextAlignment = owner.itemsTextAlignment;
```
Comment (PeterStaev) [inline drop-down.ios.ts]: change naming to be consistent with the rest of the code. So this should be just `itemsTextAlignment`

File: drop-down.ios.ts [hunk 10]
```diff
@@ -362,8 +370,11 @@ class DropDownListPickerDelegateImpl extends NSObject implements UIPickerViewDel
...
+        if (p_itemsTextAlignment === "") {
```
Comment (PeterStaev) [inline drop-down.ios.ts]: compare with `"initial"`

**Comment by AWhiteSheep** [comment]
Dropping the merge all fixed code has been pushed to the main branch.

**Pull request closed without merging**